### PR TITLE
Product QA | Functional Automation - LPS-185995 Use dynamic filters defined in fragment

### PR DESCRIPTION
**Pull request by marcosvictor22**
**JIRA Ticket:**

https://liferay.atlassian.net/browse/LPS-195333

### CODE REVIEW

**Comment by dsanz** [comment]
Hey @marcosvictor22 welcome to the team!

FYI, here in the repo, we use to provide meaningful titles to the PRs, normally, we do so by including the full LPS title/summary in the PR title, or, if the LPS is a tech task, we might use the parent task LPS title instead.

This way, just by looking at the list of PR we know, whether it's related to testing, bugfixing, tech-debt or new features.

Would you mind changing the PR title to reflect its purpose?

Thanks a lot!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 1]
```diff
@@ -1107,6 +1107,577 @@ definition {
...
+                key_radioOption = "Multiple",
+                locator1 = "CommerceEntry#RADIO_BUTTON");
+
+            Click.clickAtNotVisible(
+                key_radioOption = "Include",
+                locator1 = "CommerceEntry#RADIO_BUTTON");
+        }
+        task ("And the user save the change clicking on save button") {
+            Button.clickSave();
+        }
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1269]: this should be part of the previous task, as you are just finishing the filter creation

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 2]
```diff
@@ -1107,6 +1107,577 @@ definition {
...
+
+            Click.javaScriptClick(locator1 = "WorkflowMetricsIndex#METRICS_REINDEX_ALL_BUTTON");
+        }
+
+        task ("Then the data set fragment pagination is according to the default configuration") {
+
+            Navigator.openURL();
+
+            Navigator.gotoPage(pageName = "Test Page");
+        }
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1308]: Here you are saying that you are checking the pagination but:
1. You don't have to check the pagination, you have to check that filter is working
2. In the task you are not checking anything, you are just going to the page

Please add the assert checking the filter is present and working (Include option is selected and the preselected values are present)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 3]
```diff
@@ -1107,6 +1107,577 @@ definition {
...
+                key_labelName = "Transport",
+                locator1 = "CreateObject#OBJECT_NAME");
+
+            Button.clickSave();
+
+            ObjectAdmin.goToFieldsTab();
+
+            Refresh();
+
+		 }
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1353]: This should be another task saying that you are creating an object

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 4]
```diff
@@ -1107,6 +1107,577 @@ definition {
...
+            Navigator.openURL();
+
+            Navigator.gotoPage(pageName = "Test Page");
+        }
+    }
+
+	@description = "LPS-195333. Allow users to use defined filters when viewing the dataset they selected"
+    @priority = 5
+    test CheckingTheFilterValue {
+        task ("Given that the user added a Pickilist") {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1314]: this should say Picklist

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 5]
```diff
@@ -1107,6 +1107,577 @@ definition {
...
+                dataSetName = "Dataset Test",
+                dataSetViewDescription = "FDSViewDescription",
+                key_dataSetViewNameList = "View Fields Test");
+
+            DataSetAdmin.goToSpecificViewPage(
+                dataSetName = "Dataset Test",
+                key_viewName = "View Fields Test");
+        }
+
+        task ("When goes to Fields Tab.") {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1406]: This should say Filters tab and also you are doing more things than just going to the Filters tab, you are creating the filter itself

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 6]
```diff
@@ -1107,6 +1107,577 @@ definition {
...
+            Click.javaScriptClick(locator1 = "WorkflowMetricsIndex#METRICS_REINDEX_ALL_BUTTON");
+        }
+
+        task ("Then the data set fragment pagination is according to the default configuration") {
+
+            Navigator.openURL();
+
+            Navigator.gotoPage(pageName = "Test Page");
+        }
+    }
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1500]: Here you are saying that you are checking the pagination but:

You don't have to check the pagination, you have to check that filter is working
In the task you are not checking anything, you are just going to the page
Please add the assert checking the filter is present and working (Exclude option is selected and the preselected values are present)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 7]
```diff
@@ -1107,6 +1107,577 @@ definition {
...
+
+        task ("And Select the option Multiple,Include radio button") {
+            Click.clickAtNotVisible(
+                key_radioOption = "Multiple",
+                locator1 = "CommerceEntry#RADIO_BUTTON");
+
+            Click.clickAtNotVisible(
+                key_radioOption = "Include",
+                locator1 = "CommerceEntry#RADIO_BUTTON");
+        }
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1454]: Here you have to select the Exclude option

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 8]
```diff
@@ -1107,6 +1107,577 @@ definition {
...
+
+        task ("Then the data set fragment pagination is according to the default configuration") {
+
+            Navigator.openURL();
+
+            Navigator.gotoPage(pageName = "Test Page");
+        }
+    }
+
+	@description = "LPS-195333. Allow users to not use any dynamic filters defined in the dataset view in the shard"
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1502]: Same here, in the description we use the LPS parent story. So in this case, this should be LPS-185995. Please change it. Also, the description is not correct. In this test we are validating that, using pre-populated values, Single mode and the option Exclude, we can see the filter in the fragment.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 9]
```diff
@@ -1107,6 +1107,577 @@ definition {
...
+    @priority = 5
+    test CanViewWithoutFilters {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1504]: In this test we are validating that, using pre-populated values and the option Exclude, we can see the filter in the fragment, so the name of this test cannot be something that generic. It should be something like "CheckDynamicFilterInFragmentWithSinglePrePopulatedValueAndExclude"

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 10]
```diff
@@ -1107,6 +1107,577 @@ definition {
...
+                key_labelName = "Transport",
+                locator1 = "CreateObject#OBJECT_NAME");
+
+            Button.clickSave();
+
+            ObjectAdmin.goToFieldsTab();
+
+            Refresh();
+
+		 }
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1544]: This should be another task saying that you are creating an object

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 11]
```diff
@@ -1107,6 +1107,577 @@ definition {
...
+            Navigator.openURL();
+
+            Navigator.gotoPage(pageName = "Test Page");
+        }
+    }
+
+	@description = "LPS-195333. Allow users to not use any dynamic filters defined in the dataset view in the shard"
+    @priority = 5
+    test CanViewWithoutFilters {
+        task ("Given that the user added a Pickilist") {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1505]: This should say Picklist

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 12]
```diff
@@ -1107,6 +1107,577 @@ definition {
...
+                dataSetName = "Dataset Test",
+                dataSetViewDescription = "FDSViewDescription",
+                key_dataSetViewNameList = "View Fields Test");
+
+            DataSetAdmin.goToSpecificViewPage(
+                dataSetName = "Dataset Test",
+                key_viewName = "View Fields Test");
+        }
+
+        task ("When goes to Fields Tab.") {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1598]: This should say Filters tab and also you are doing more things than just going to the Filters tab, you are creating the filter itself

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 13]
```diff
@@ -1107,6 +1107,577 @@ definition {
...
+
+            Select(
+                locator1 = "DataSet#SOURCE_OPTIONS_SELECT",
+                value1 = "Transport");
+        }
+
+        task ("And Select the option Multiple,Include radio button") {
+            Click.clickAtNotVisible(
+                key_radioOption = "Multiple",
+                locator1 = "CommerceEntry#RADIO_BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1631]: This should select the Single option and then select just one pre populated value from the picklist

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 14]
```diff
@@ -1107,6 +1107,577 @@ definition {
...
+
+            Click.clickAtNotVisible(
+                key_radioOption = "Include",
+                locator1 = "CommerceEntry#RADIO_BUTTON");
+        };
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1636]: This should be Exclude, not include

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 15]
```diff
@@ -1107,6 +1107,577 @@ definition {
...
+
+            Click.javaScriptClick(locator1 = "WorkflowMetricsIndex#METRICS_REINDEX_ALL_BUTTON");
+        }
+
+        task ("Then the data set fragment pagination is according to the default configuration") {
+
+            Navigator.openURL();
+
+            Navigator.gotoPage(pageName = "Test Page");
+        }
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1681]: Here you are saying that you are checking the pagination but:

You don't have to check the pagination, you have to check that filter is working
In the task you are not checking anything, you are just going to the page
Please add the assert checking the filter is present and working (Exclude option is selected and the preselected values are present)

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 16]
```diff
@@ -1107,6 +1107,577 @@ definition {
...
+                key_value = "Select Data Set View",
+                locator1 = "Button#SECONDARY_BY_TITLE");
+            
+			SelectFrame(
+                key_title = "Select",
+                locator1 = "IFrame#MODAL_ANY");
+            
+			Click.clickAtNotVisible(
+                key_radioOption = "View Fields Test",
+                locator1 = "CommerceEntry#RADIO_BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1665]: This Commerce path has nothing to do with our test. Please use a generic button path from the Button.path file

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 17]
```diff
@@ -1107,6 +1107,577 @@ definition {
...
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1483]: This Commerce path has nothing to do with our test. Please use a generic button path from the Button.path file

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase [hunk 18]
```diff
@@ -1107,6 +1107,577 @@ definition {
...
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSet.testcase:1292]: This Commerce path has nothing to do with our test. Please use a generic button path from the Button.path file

**Review by CarlosBrichete — CHANGES_REQUESTED** [review]
Please check all my requests. @deborasoliveira maybe you can help him to understand better this automation

**Comment by CarlosBrichete** [comment]
@marcosvictor22 Also please add a title and a description to this PR, so anyone can have a clear vision of the purpose of this PR, thanks :)

**Comment by mateusralv** [comment]
After a fats analysis I add the modifications: 
- [ ] Add the Names of the tests and the name of the test file to the jira ticket 
- [ ] Add a Description to the PR

**Comment by marcosvictor22** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetFragmentFilters.testcase [hunk 19]
```diff
@@ -217,6 +217,481 @@ definition {
...
+		}
+
+		task ("And Select the option Multiple,Include radio button") {
+			Click.clickAtNotVisible(
+				key_radioOption = "Multiple",
+				locator1 = "CommerceEntry#RADIO_BUTTON");
+
+			Click.clickAtNotVisible(
+				key_radioOption = "Include",
+				locator1 = "CommerceEntry#RADIO_BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetFragmentFilters.testcase:530]: You should not use a Commerce path. Use a generic radio button path please

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetFragmentFilters.testcase [hunk 20]
```diff
@@ -217,6 +217,481 @@ definition {
...
+		}
+
+		task ("When the user creates a field guide and goes to it") {
+			DataSetAdmin.goToTab(tabName = "Filters");
+
+			LexiconEntry.gotoAdd();
+
+			Click(
+				key_fieldName = "Selection",
+				locator1 = "FormViewBuilder#OBJECT_FIELD_BUTTON_DROPDOWN");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetFragmentFilters.testcase:323]: You should not use a Form path. Use a generic dropdown path please

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetFragmentFilters.testcase [hunk 21]
```diff
@@ -217,6 +217,481 @@ definition {
...
+
+			Click(
+				key_sortOption = "bike",
+				locator1 = "ObjectCustomViews#SORT_OPTION");
+		}
+
+		task ("And Select the option Multiple,Include radio button") {
+			Click.clickAtNotVisible(
+				key_radioOption = "Multiple",
+				locator1 = "CommerceEntry#RADIO_BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetFragmentFilters.testcase:359]: You should not use a Commerce path. Use a generic radio button path please

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetFragmentFilters.testcase [hunk 22]
```diff
@@ -217,6 +217,481 @@ definition {
...
+
+			Click.clickAtNotVisible(
+				key_radioOption = "Exclude",
+				locator1 = "CommerceEntry#RADIO_BUTTON");
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetFragmentFilters.testcase:363]: You should not use a Commerce path. Use a generic radio button path please

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetFragmentFilters.testcase [hunk 23]
```diff
@@ -217,6 +217,481 @@ definition {
...
+				dataSetName = "Dataset Test",
+				dataSetViewDescription = "FDSViewDescription",
+				key_dataSetViewNameList = "View Fields Test");
+
+			DataSetAdmin.goToSpecificViewPage(
+				dataSetName = "Dataset Test",
+				key_viewName = "View Fields Test");
+		}
+
+		task ("When the user creates a field guide and goes to it") {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetFragmentFilters.testcase:475]: What's a field guide? The user is creating a dynamic filter

**Review by CarlosBrichete — CHANGES_REQUESTED** [review]
Please check my requested changes

**Pull request closed without merging**